portal-impl/src/com/liferay/portal/image/ImageToolImpl.java

```diff
@@ -316,7 +322,53 @@ public RenderedImage crop(
 	public byte[] cropGif(
 		byte[] imageData, int height, int width, int x, int y) {
 
-		return null;
+		ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+		try {
+			ImageOutputStream ios = ImageIO.createImageOutputStream(os);
+
+			ImageInputStream imageInputStream = ImageIO.createImageInputStream(
+				new ByteArrayInputStream(imageData));
+
+			ImageReader imageReader = (ImageReader)
+				ImageIO.getImageReadersByFormatName(ImageTool.TYPE_GIF).next();
+
+			ImageWriter imageWriter = (ImageWriter)
+				ImageIO.getImageWritersByFormatName(ImageTool.TYPE_GIF).next();
+
+			imageReader.setInput(imageInputStream, false);
+
+			imageWriter.setOutput(ios);
+			imageWriter.prepareWriteSequence(null);
+
+			IIOMetadata imageMetaData = imageReader.getImageMetadata(0);
+			int numberOfFrames = imageReader.getNumImages(true);
+			BufferedImage master = null;
+
+			for (int i = 0; i < numberOfFrames; i++) {
+				BufferedImage image = imageReader.read(i);
+
+				master = new BufferedImage(
+					width, height, BufferedImage.TYPE_INT_ARGB);
+
+				master.getGraphics().drawImage(image, x, y, null);
+
+				IIOImage iioImage = new IIOImage(master, null, imageMetaData);
+
+				imageWriter.writeToSequence(iioImage, null);
+			}
+
+			int delay = _getDelay(imageMetaData);
```

Comment:
@hudakl 

What is the porpouse of this call?? This variable is never used.

The imageMetaData is been modified in this method?? If not I do not understand why do we have to do this call.

Thanks!

Reply:
@hudakl 

Did you see this??

Reply:
Hi,

I just saw it, because I was on holiday on Friday. I added that because without the delay time the gif wasn't played at the proper speed, but now I indeed can't find where it is actually used.

I doublecheck and get back to you.

Laci